packages/mysql/lib/mysql_p.js

```diff
@@ -141,12 +141,7 @@ function resolveArguments(argsObj) {
 
   if (argsObj && argsObj.length > 0) {
     if (argsObj[0] instanceof Object) {
-      args.sql = argsObj[0].sql;
-
-      if (argsObj[0].timeout) {
-        args.timeout = argsObj[0].timeout;
-      }
-
+      args.sql = argsObj[0];
```

Comment:
Doesn't this set `args.sql` to an entire argument object rather than just the query string as it was before? I don't have any problem with this, I'm just confused by how the same behavior is maintained. Specifically want to make sure that calling the original operation will still work if someone is using `mysql` in the mode captured by the `else` block, where they pass the query string as the first parameter rather than an arguments object.

Reply:
@willarmiros that's correct, since the [original `query` function](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/mysql/index.d.ts#L384) accepts an object as its first argument, this just ensures that the entire object does in fact get passed along (vs. just the `sql` and `timeout` properties). That's a fair concern though; this condition could definitely be made simpler and easier to reason about now that we're passing along the entire object, however I decided to leave things as-is for now so as not to overcomplicate this PR. The logic in the `else` block seems like it should support the query string flavor just fine though, and the tests all passed w/o issue. 👍 Either way, let me know if this needs any clarification; happy to discuss further.